### ISSUE TITLE
[Question] MapAlignerTreeGuided transformation statistics "too good"?

### ISSUE THREAD

**Issue by jpfeuffer** [OpenMS/OpenMS#5038]
Hi (especially @jthueringer ),

I have a question about the transformations produced in MATG:
I implemented an align method to be used from other tools using the "copy" strategy of the tool here:
https://github.com/jpfeuffer/OpenMS/blob/d3661bdadd296f5e83280c00262e8817609f940f/src/openms/source/ANALYSIS/MAPMATCHING/MapAlignmentAlgorithmTreeGuided.cpp#L269-L295

When I apply this method to my FeatureMaps and then evaluate the statistics of the resulting (global?) Transformations like this:
https://github.com/jpfeuffer/OpenMS/blob/d3661bdadd296f5e83280c00262e8817609f940f/src/utils/ProteomicsLFQ.cpp#L570-L581

I always get very good transformations:
```
100% 99% 95% 90% 75% 50% 25%
before alignment:
133 61 61 61 57 45 22
after alignment:
4 1 0 0 0 0 0 
```

Or sometimes even better (0 everywhere). How can this be explained?

**Comment by jthueringer** [comment]
Hi @jpfeuffer ,

today I finally managed to deal with your question. I am not quite sure if I understood the problem correctly, but maybe the following is the answer:
First of all, I assume that your example is the result for one map, but you get the statistics for each map?!

Explanation attempt: During the alignment the transformations determined by MA-Identification are already applied. During the process all maps to be aligned are combined into one. After that the MATG determines the transformed values for each identification based on the resulting last transformed map and creates new TransformationDescriptions from these values. Thus the values are not global and the transformation results look so good. If you get 0 everywhere it only says that there was no error produced by seperating the resulting map that contained all aligned maps and extract the TransformationDescriptions for each input map.

I hope my explanation is understandable and answers your question. If not, feel free to contact me again.

(I think more informative is to analyse the alignment quality after feature linking, eg. using the tool PTXQC)

**Comment by jpfeuffer** [comment]
Does this mean, that the transformations that you get out are only the transformations from the "last step" in the tree alignment? I think in other map aligners we report the transformations needed to go from the very initial input feature locations to the consensus point that was finally determined by the alignment algorithm.
I am not sure if I completely understood the explanation though 🙂

**Comment by jthueringer** [comment]
Yes, it's correct. It's a report about the transformations needed to go from the very initial input feature location to the point that was calculated from the transformation model within the last step in the tree alignment (depending on the consensus). We (@cbielow ) decided that it's okay to choose this way, because the deviations in the resulting transformed maps are only small.
Should it be adapted to what the other aligners report or does it work as it is?

**Comment by timosachsenberg** [comment]
To select proper parameters for e.g. the linking we would prefer to have the error reported the same way if possible. Currently, it doesn't play well with existing code so it is a bit inconvenient to plug in your alignment algorithm (which we would love to do) e.g., as a substitute to the existing MapAlingerIdentification one in the ProteomcisLFQ tool.

**Comment by jthueringer** [comment]
I see. 
A new challenge ;)

**Comment by timosachsenberg** [comment]
Hehe, I don't have any clue if that is easy to implement or not :P but we would be really grateful if we would get some help here.

**Comment by jthueringer** [comment]
I try to find a way and let you know if it doesn't work - or with a new PR.

**Comment by jthueringer** [comment]
It's really not easy, because even if you figure out which map the respective TransformationDescription belongs to, a lot of information is lost while aligning a larger map with a smaller one.
The only idea I have is to repeat the alignments for each input-map with the resulting combined map as reference or maybe better: to first split the resulting map into the individual ones again and then perform the alignment with the corresponding counterparts.
Would that make sense to you?
Another problem is that I only use my private laptop and implement in my spare time. Hence I can't do a similarly large analysis of the results as I did earlier this year. This would be needed to estimate how much the results differ. Could one of you take care of it?

**Comment by jpfeuffer** [comment]
Yes, testing is no problem.
But I don't get why the alignment has to be repeated for the input maps.
Since we use the "copy-mode" of the algorithm, the initial feature maps should be unaffected. Can't we use those initial feature maps and compare the RT of their features with the consensus RT of the resulting map_transformed?
Here:
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/ANALYSIS/MAPMATCHING/MapAlignmentAlgorithmTreeGuided.cpp#L303

**Comment by jpfeuffer** [comment]
E.g. shouldn't this line here:
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/ANALYSIS/MAPMATCHING/MapAlignmentAlgorithmTreeGuided.cpp#L316

use the RT from the input map? And not the one from the transformed map?

**Comment by jthueringer** [comment]
Now I got it. You told me that you need the RT of the consensus feature created, but I didn't think about it :facepalm: 

The metaValue "original_RT" is written once and not overwritten so it can be used. 

Many many thanks for your answer!

**Comment by jpfeuffer** [comment]
By the way, are you on gitter? https://gitter.im/OpenMS

Maybe we could chat, then it might be quicker with the understanding ;)

**Comment by timosachsenberg** [comment]
Has this been implemented?

**Comment by jpfeuffer** [comment]
Hmm now I am not sure anymore.

**Comment by timosachsenberg** [comment]
@cbielow do you know? otherwise I would close issue. inactive for several years
